federation-2/router-bridge/src/worker.rs

```diff
@@ -67,26 +87,73 @@ where
         Self {
             sender,
             handle: Some(handle),
-            receiver,
+            response_receivers: Default::default(),
+            response_senders,
         }
     }
 
-    pub(crate) async fn request(&self, command: Request) -> Result<Response, Error> {
-        self.sender
+    pub(crate) async fn request<Request, Response>(
+        &self,
+        command: Request,
+    ) -> Result<Response, Error>
+    where
+        Request: Serialize + Send + Debug + 'static,
+        Response: DeserializeOwned + Send + Debug + 'static,
+    {
+        let id = self
             .send(command)
             .await
             .map_err(|e| Error::DenoRuntime(format!("couldn't send request {e}")))?;
-        self.receiver
-            .recv()
+        self.receive(id)
             .await
             .map_err(|e| Error::DenoRuntime(format!("request: couldn't receive response {e}")))
     }
 
-    pub(crate) async fn send(&self, command: Request) -> Result<(), Error> {
+    pub(crate) async fn send<Request>(&self, request: Request) -> Result<Uuid, Error>
+    where
+        Request: Serialize + Send + Debug + 'static,
+    {
+        let id = Uuid::new_v4();
+
+        let (sender, receiver) = mpsc::channel(1);
```

Comment:
a oneshot channel should be enough here

Reply:
Oh indeed, it's a leftover from my dashmap tryouts!

Reply:
fixed in f47bac8